astroquery/esa/hubble/core.py

```diff
@@ -215,6 +219,85 @@ def get_observation_type(self, observation_id):
             raise ValueError("Invalid Observation ID")
         return obs_type
 
+    def get_observations_from_program(self, program, *, output_file=None,
+                                      output_format="votable", verbose=False):
+        """
+        Retrieves all the observations associated to a program/proposal ID
+
+        Parameters
+        ----------
+        program : int
+            Program or Proposal ID associated to the observations
+        output_file : string
+            optional, default None
+            file name to be used to store the result
+        output_format : string
+            results format. Options are:
+            'votable': str, binary VOTable format
+            'csv': str, comma-separated values format
+        verbose : bool
+            optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        None. It downloads the artifact indicated
+        """
+        return self.query_criteria(proposal=program, output_file=output_file,
+                                   output_format=output_format, async_job=False, verbose=verbose)
+
+    def download_files_from_program(self, program, *, folder=os.getcwd(), calibration_level=None,
```

Comment:
What about using a more generic name, such as `download_files` or `download_products` and making `program` optional? (e.g. allowing the download of any results of `query_criteria`, as I see the only difference in the signature is that this mandates `program` to be specified, while it's optional for `query_criteria`)

Reply:
Same as before, the wanted an explicit method. If it is not ok with you, please let me know.

Reply:
Overall I think it's better to try not to have too many methods that only differ minimally and use more parameters if needed (good for removing code duplications, etc.). However, it's always difficult to say where the line is especially if there is already a user request for a functionality.
So it's all good for now.